sources/iTermApplicationDelegate.m

```diff
@@ -1440,12 +1440,21 @@ - (void)nonTerminalWindowBecameKey:(NSNotification *)aNotification {
     [closeWindow setKeyEquivalentModifierMask:NSEventModifierFlagCommand];
 }
 
-- (void)updateAddressBookMenu:(NSNotification*)aNotification {
-    DLog(@"Updating address book menu");
+- (void)updateAddressBookMenu:(NSNotification *)aNotification {
+    DLog(@"Updating Profile menu");
     JournalParams params;
-    params.selector = @selector(newSessionInTabAtIndex:);
+    if ([iTermAdvancedSettingsModel openProfilesInNewWindow]) {
+        params.selector = @selector(newSessionInWindowAtIndex:);
+        params.alternateSelector = @selector(newSessionInTabAtIndex:);
+        [bookmarkMenu itemAtIndex:2].title = [[bookmarkMenu itemAtIndex:2].title stringByReplacingOccurrencesOfString:@"Window" withString:@"Tab"];
+        [bookmarkMenu itemAtIndex:3].title = [[bookmarkMenu itemAtIndex:3].title stringByReplacingOccurrencesOfString:@"Window" withString:@"Tab"];
```

Comment:
I think you'll also want to update the last item, which says "Open All in New Window" when holding option.

Reply:
So right now, that item still works the old way - will do tabs unless you hold Option.  Think it makes sense to flip it as well?

Reply:
After taking a second look, I think it's OK to keep it as is. I'll merge this.